### PR TITLE
properties helper, additional api methods

### PR DESCRIPTION
**Pull request by atheiman**
- [x] hash to Java .properties file
  - [x] specs
- [x] api client post, put, delete functionality
  - [x] specs
  - [x] kitchen tests
- [x] api client logger
  - [x] specs

### CODE REVIEW

**Comment by atheiman** [comment]
this is ready for review

**Comment by atheiman** [comment]
@pcross616 can you review this?

File: libraries/rundeck_api_client.rb [hunk 1]
```diff
@@ -50,23 +51,61 @@ def send_req(req)
...
+  end
+
+  def delete(path)
+    request(Net::HTTP::Delete, path)
+  end
 
   def parse_res(res)
-    JSON.parse(res.body)
+    # TODO: make this decision based on Content-Type of response
+    if res.body.to_s.empty? || res.body.to_s.start_with?('<!DOCTYPE html>')
```
Comment (pcross616) [inline libraries/rundeck_api_client.rb]: Why cant we just read the content.  We have the reaponse vs the string check its doing now.  Seems brittle.

Comment (atheiman) [inline libraries/rundeck_api_client.rb]: Ill make this update today

Comment (atheiman) [inline libraries/rundeck_api_client.rb]: updated logic to use `Content-Type` header

**Comment by atheiman** [comment]
this is ready for review again @ARentz07 @pcross616

**Pull request merged**